### PR TITLE
[AzureMonitorExporter] update OpenTelemetry dependencies to v1.5.0

### PR DESCRIPTION
**Pull request by TimothyMothra**
### Changes
- update OpenTelemetry dependencies to v1.5.0
- Breaking changes: 
  - `LogRecord.StateValues` replaced with `LogRecord.Attributes`
  - `ParseStateValues` will be ignored unless logging an unknown object.
  - see also: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334
  - AzureMonitorExporterLoggingExtensions will no longer set ParseStateValues to true.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterLoggingExtensions.cs [hunk 1]
```diff
@@ -31,10 +31,6 @@ public static OpenTelemetryLoggerOptions AddAzureMonitorLogExporter(this OpenTel
                 throw new ArgumentNullException(nameof(loggerOptions));
             }
 
-            // Ideally user should set this to true
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterLoggingExtensions.cs:34]: There are many tests that uses `ParseStateValues`. Should we remove those tests?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterLoggingExtensions.cs:34]: Let me check and see what the impact is

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterLoggingExtensions.cs:34]: This is safe to remove from most of our tests. We had two tests that were asserting true & false. I combined that into one test using `Theory`

**Comment by vishweshbankwar** [comment]
@TimothyMothra It would be good to validate the changes against otlp exporter changes that was done in the pr you referenced.

**Comment by TimothyMothra** [comment]
> @TimothyMothra It would be good to validate the changes against otlp exporter changes that was done in the pr you referenced.

I don't see any of the Exporters in the otel repo checking the `ParseStateValues`

**Comment by TimothyMothra** [comment]
TODO: Need to lock to SDK v1.5.x or lower because of coming breaking changes.

**Comment by cijothomas** [comment]
> > @TimothyMothra It would be good to validate the changes against otlp exporter changes that was done in the pr you referenced.
> 
> I don't see any of the Exporters in the otel repo checking the `ParseStateValues`

There are no breaking changes. So even if this exporter do nothing, everything would work as before. (Ofcourse there will be warning about using obsolete things, but if you suppress it, then there is no need of any change here.)

**Comment by cijothomas** [comment]
> TODO: Need to lock to SDK v1.5.x or lower because of coming breaking changes.

Where is breaking change in Otel sdk?

**Comment by TimothyMothra** [comment]
> > TODO: Need to lock to SDK v1.5.x or lower because of coming breaking changes.
> 
> Where is breaking change in Otel sdk?

Breaking changes are coming in the Instrumentation Libraries. We want to snap the Exporter to a known version of the base SDK.

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -120,13 +120,13 @@
 
   <ItemGroup Condition="$(MSBuildProjectName.StartsWith('Azure.Monitor.OpenTelemetry'))">
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
-    <PackageReference Update="OpenTelemetry" Version="1.4.0" />
-    <PackageReference Update="OpenTelemetry.Exporter.InMemory" Version="1.4.0" />
+    <PackageReference Update="OpenTelemetry" Version="[1.5.0, 1.6.0)" />
```
Comment (TimothyMothra) [inline eng/Packages.Data.props:123]: Accepts version 1.5.0 up to 1.5.x on update, but not 1.6 and higher.

The Instrumentation Libraries are implementing breaking changes wrt Http Semantic Conventions. We want to use this to lock the Exporter to the v1.5.0 milestone.

Comment (vishweshbankwar) [reply eng/Packages.Data.props:123]: I am not sure why this is needed.

Comment (rajkumar-rangaraj) [reply eng/Packages.Data.props:123]: I recommend to keep the scope of this PR to just upgrade the OpenTelemetry SDK without introducing additional dependencies. 

You could propose this change in a follow up PR.

Comment (TimothyMothra) [reply eng/Packages.Data.props]: reverted this change

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM.

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs [hunk 3]
```diff
@@ -24,6 +24,9 @@ public class LogsHelperTests
         [InlineData(false)]
         public void MessageIsSetToExceptionMessage(bool parseStateValues)
         {
+            // ParseStateValues will be ignored unless the log contains an unknown objects.
```
[the remark this answers is not in the document]
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs:27]: We should investigate this more. I wasn't able to reproduce the behavior that Blanch described in his PR.

**Pull request merged**